### PR TITLE
[70539] Validate access_token_user_attributes for service_account_configs

### PR DESCRIPTION
**Pull request by dickdavis**
## Summary

This PR validates the contents of the `access_token_user_attributes` field to ensure that only allowed values are set.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#70539

## Testing done

- Use the Rails console to load a service account config.
```ruby
config = SignIn::ServiceAccountConfig.last
```
- Attempt to store an invalid value (`['foobar'] }`)
```ruby
config.access_token_user_attributes = ['foobar']
```
- Verify the validation fails.
- Attempt to store a valid value (`['icn']`)
```ruby
config.access_token_user_attributes = ['icn']
```
- Verify the validation succeeds.

## Screenshots
Not relevant.

## What areas of the site does it impact?
This PR impacts service account authentication.

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by bramleyjl — APPROVED** [review]

File: spec/factories/sign_in/service_account_configs.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@
     scopes { [Faker::Internet.url] }
     access_token_audience { SecureRandom.hex }
     access_token_duration { SignIn::Constants::AccessToken::VALIDITY_LENGTH_SHORT_MINUTES }
```
Comment (bosawt) [inline spec/factories/sign_in/service_account_configs.rb:9]: We should use your new constant for this field

File: app/services/sign_in/constants/service_account_access_token.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module SignIn
+  module Constants
+    module ServiceAccountAccessToken
```
Comment (bosawt) [inline app/services/sign_in/constants/service_account_access_token.rb:5]: We might as well go ahead and copy the rest of the access token constants into this and refer to this set of constants wherever we use the service account functionality

If you do a global search for `Constants::AccessToken` you should be able to see everywhere that needs to be changed to the service account access token without too much difficulty (I only see ~50 usages so not too huge)

**Review by bosawt — CHANGES_REQUESTED** [review]
Mostly looks good, just some simple changes around using the constants more consistently.

**Review by kpethtel — APPROVED** [review]
Good AFAICT

**Review by bramleyjl — DISMISSED** [review]
Thanks for updating the service account constants!

**Review by bosawt — APPROVED** [review]
looks gud. Tested by attempting to create a ServiceAccount Config with user attributes that aren't `ICN` (which properly failed), and attempting to create a ServiceAccountConfig with `ICN` user attribute (which worked)

**Review by rjohnson2011 — APPROVED** [review]

**Pull request merged**